### PR TITLE
feat(NODE-4815): stringify and truncate BSON documents in log messages

### PR DESCRIPTION
**Pull request by W-A-James**
#### What is changing?

Adds the `maybeTruncate` helper and calls it in the `defaultLogTransform` helper function to produce spec-compliant logs.

### Double check the following

- [x] Ran `npm run check:lint` script
- [x] Self-review completed using the [steps outlined here](https://github.com/mongodb/node-mongodb-native/blob/HEAD/CONTRIBUTING.md#reviewer-guidelines)
- [x] PR title follows the [correct format](https://www.conventionalcommits.org/en/v1.0.0/): `type(NODE-xxxx)[!]: description`
  - Example: `feat(NODE-1234)!: rewriting everything in coffeescript`
- [x] Changes are covered by tests
- [x] New TODOs have a related JIRA ticket

### CODE REVIEW

File: src/mongo_logger.ts [hunk 1]
```diff
@@ -292,22 +306,23 @@ function attachConnectionFields(
 }
 
 function defaultLogTransform(
-  logObject: LoggableEvent | Record<string, any>
+  logObject: LoggableEvent | Record<string, any>,
+  maxDocumentLength: number = DEFAULT_MAX_DOCUMENT_LENGTH
```
Comment (nbbeeken) [inline src/mongo_logger.ts:310]: We always will default this in the options do we need to redo that here and in truncation?

File: src/mongo_logger.ts [hunk 2]
```diff
@@ -254,6 +256,18 @@ export interface LogConvertible extends Record<string, any> {
   toLog(): Record<string, any>;
 }
 
+/** @internal */
+export function maybeTruncate(
```
Comment (nbbeeken) [inline src/mongo_logger.ts:260]: Can we move EJSON stringify calls into the same function that worries about enforcing the size limit? maybeTruncate -> stringifyWithLength (or whatever name seems cooler)

File: src/mongo_logger.ts [hunk 3]
```diff
@@ -254,6 +256,18 @@ export interface LogConvertible extends Record<string, any> {
...
+  ejson: string,
+  maxDocumentLength: number = DEFAULT_MAX_DOCUMENT_LENGTH
+): string {
+  if (maxDocumentLength === 0) {
+    return ejson;
+  }
+
+  return ejson.length > maxDocumentLength ? `${ejson.slice(0, maxDocumentLength)}...` : ejson;
```
Comment (nbbeeken) [inline src/mongo_logger.ts:268]:
```suggestion
  return maxDocumentLength !== 0 && ejson.length > maxDocumentLength ? `${ejson.slice(0, maxDocumentLength)}...` : ejson;
```

**Review by nbbeeken — CHANGES_REQUESTED** [review]

File: src/mongo_logger.ts (hunk 1 above)
Comment (W-A-James) [reply src/mongo_logger.ts:310]: Do we have a preference for where we default it? I would skew to doing it in the log transform/logger itself

Comment (nbbeeken) [reply src/mongo_logger.ts:310]: The logger class seems good

**Review by nbbeeken — APPROVED** [review]
LGTM

**Pull request merged**